base/tuple.jl

```diff
@@ -20,7 +20,7 @@ endof(t::Tuple) = length(t)
 size(t::Tuple, d) = d==1 ? length(t) : throw(ArgumentError("invalid tuple dimension $d"))
 getindex(t::Tuple, i::Int) = getfield(t, i)
 getindex(t::Tuple, i::Real) = getfield(t, convert(Int, i))
-getindex(t::Tuple, r::AbstractArray{<:Any,1}) = tuple([t[ri] for ri in r]...)
+getindex(t::Tuple, r::AbstractArray{<:Any,1}) = (n = length(eachindex(r)); ntuple(i->t[r[i]], n))
```

Comment:
Could you explain this change's purpose? Could indexing `r` with indices `1:n` (rather than iterating over `r`'s elements, or indexing `r` more genernically) be an issue? If not, why `length(eachindex(r))` rather than simply `length(r)`? Does using `ntuple` here not potentially hurt inference?